dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilder.java

```diff
@@ -108,28 +108,32 @@ private static String osDetectedArch() {
     }
   }
 
-  private static DependencyNode resolveCompileTimeDependencies(Artifact rootDependencyArtifact)
+  private static DependencyNode resolveCompileTimeDependencies(
+      DependencyNode rootDependencyArtifact)
       throws DependencyCollectionException, DependencyResolutionException {
     return resolveCompileTimeDependencies(rootDependencyArtifact, false);
   }
 
   private static DependencyNode resolveCompileTimeDependencies(
-      Artifact rootDependencyArtifact, boolean includeProvidedScope)
+      DependencyNode rootDependencyArtifact, boolean includeProvidedScope)
       throws DependencyCollectionException, DependencyResolutionException {
     return resolveCompileTimeDependencies(
         ImmutableList.of(rootDependencyArtifact), includeProvidedScope);
   }
 
   private static DependencyNode resolveCompileTimeDependencies(
-      List<Artifact> dependencyArtifacts, boolean includeProvidedScope)
+      List<DependencyNode> dependencyNodes, boolean includeProvidedScope)
       throws DependencyCollectionException, DependencyResolutionException {
 
-    Map<String, DependencyNode> cache =
+    // The cache key includes exclusion elements of Maven artifacts
+    Map<Dependency, DependencyNode> cache =
         includeProvidedScope ? cacheWithProvidedScope : cacheWithoutProvidedScope;
-    String cacheKey =
-        dependencyArtifacts.stream().map(Artifacts::toCoordinates).collect(Collectors.joining(","));
-    if (cache.containsKey(cacheKey)) {
-      return cache.get(cacheKey);
+    Dependency cacheKey = null;
+    if (dependencyNodes.size() == 1) {
```

Comment:
In our usage (dashboard and linkage monitor), the cache for multiple `dependencyArtifacts` was not used.